src/resources/assets/js/store/actions.js

```diff
@@ -161,6 +161,19 @@ export default {
         path
     }) {
         commit(types.SET_LOADING, true)
+
+        let cleanUp = () => {
+            commit(types.SET_LOADING, false);
+        }
+
+        let fileType = type === 'file' ? 'new file' : 'new folder';
+
+        if (path.length === 0) {
+            alertify.error('Please provide a name for the ' + fileType);
+            cleanUp();
+            return;
+        }
+
         path = withPwd(state, path)
 
         api.create(type, path).then(() => {
```

Comment:
There is a `catch` block on this call, please add a `cleanUp()` on it to it to avoid being stuck when an error happens:

```js
api.create(type, path).then(() => {
    let newtype = type === 'file' ? 'New file' : 'New folder'
    alertify.success(newtype + ' created')
    dispatch(types.REFRESH)
}).catch(e => {
    alertify.error('Failed to create ' + type)
    cleanUp()
})
```